src/lib/utils/aura-rewards.ts

```diff
@@ -0,0 +1,212 @@
+import * as pMemoize from 'p-memoize';
+import { MinParaBoostData, fetchAccountsScores } from './staking-supervisor';
+import BigNumber from 'bignumber.js';
+import { GasRefundV2EpochFlip } from '../gas-refund/gas-refund';
+import { getCurrentEpoch } from '../gas-refund/epoch-helpers';
+import { CHAIN_ID_OPTIMISM, STAKING_CHAIN_IDS } from '../constants';
+import {
+  AddressRewards,
+  AddressRewardsWithAmountsByProgram,
+} from '../../types';
+
+const config: Record<number, string> = {
+  // @TODO: update this config
+  41: '69311354102943179612461',
+};
+
+const AURA_REWARDS_START_EPOCH_OLD_STYLE = Math.min(
+  ...Object.keys(config).map(Number),
+);
+
+const logger = global.LOGGER('aura-rewards');
+
+async function _fetchEpochData(epoch: number) {
+  const list = await fetchAccountsScores(epoch);
+
+  const byAccountLowercase = list.reduce<Record<string, MinParaBoostData>>(
+    (acc, curr) => {
+      acc[curr.account.toLowerCase()] = curr;
+      return acc;
+    },
+    {},
+  );
+
+  logger.info(
+    `loaded pre-requisites for computing Aura rewards for epoch ${epoch}`,
+  );
+
+  // computing total with BigNumber instead of taking it from the endpoint
+  const totalScore = list
+    .reduce((acc, curr) => {
+      return acc.plus(curr.score);
+    }, new BigNumber(0))
+    .toFixed();
+
+  return {
+    totalScore,
+    list,
+    byAccountLowercase,
+  };
+}
+
+// cache forever
+const fetchPastEpochData = pMemoize(_fetchEpochData, {
+  cacheKey: ([epoch]) => `paraboost_epochData_${epoch}`,
+});
+
+// is used during when executing distribution script.
+// is intended to compute user rewards for the current epoch without remainder
+export async function computeUserRewardWei(
+  user: string,
+  epochOldStyle: number,
+  counter: RewardsDistributionCounter,
+) {
+  if (epochOldStyle == getCurrentEpoch()) {
+    // not known yet - hence display 0
+    return '0';
+  }
+
+  // for epochs before Aura rewards started, return 0
+  if (epochOldStyle < AURA_REWARDS_START_EPOCH_OLD_STYLE) return '0';
+
+  const epoch = epochOldStyle - GasRefundV2EpochFlip;
+  const { byAccountLowercase, totalScore } = await fetchPastEpochData(epoch);
+
+  const userScore = byAccountLowercase[user.toLowerCase()]?.score || '0';
+
+  const totalRewards = config[epochOldStyle] || '0';
+  const remainingRewards = new BigNumber(totalRewards)
+    .minus(counter.rewardsAllocated.toString())
+    .toFixed();
+  const remainingTotalScore = new BigNumber(totalScore)
+    .minus(counter.scoreCleared.toString())
+    .toFixed();
+  const userRewards = new BigNumber(userScore)
+    .times(remainingRewards)
```

Comment:
Injecting `counter` to keep track of already deducted rewards is a bit convoluted. What was the problem with `userRewards=auraRewards*userScore/totalScore`? numbers didn't add up?

Reply:
`sum != allocated amount`, yes

i'm not very happy with this approach too, particularly because I think pre-ordering does matter.
 
how'd you approach though?